Added guidelines on how to code components (#571)
SE-UUlm/snowballr-frontend: wiki/Contributing.md

│   │   └── model/
│   │       └── api/ (auto-generated API code)
│   └── routes/ (website layout)
├── tests/
│   ├── e2e/
│   ├── integration/
│   └── unit/
└── wiki/ (this wiki)
```

## Creating a New Component

When creating a new component, please follow these guidelines:

1. **Location**: Place your component in the appropriate directory under `src/lib/components/`. Use `composites/` for
   custom components and `primitives/` for shadcn/ui components (refer to
   [the installation guide](https://shadcn-svelte.com/docs/installation/sveltekit)).
2. **Naming**: Use PascalCase for component files (e.g., `MyComponent.svelte`) and kebab-case for normal TS files (e.g.,
   `my-helper.ts`)
3. **Props**: Clearly define the props your component accepts. Use TypeScript for type safety.

> 4. **Documentation**: Add TSDoc comments to your component and its props to explain their purpose and usage.

Reviewer: ```suggestion
4. **Documentation**: Add HTML comments followed by _@component_ to your component to explain their purpose, usage and the meaning of the props.
```
We do not use TSDoc for component comments, right?

Author: No, I removed "TSDoc". The exact way to comment on a component is explained later.

5. **Testing**: Write integration tests for your component in the `tests/integration/` directory to ensure its
   functionality (see [the testing section](#testing)).

> Example Structure for `MyNewComponent`:

Reviewer: ```suggestion
Example structure for `MyNewComponent`:
```

```svelte
<script lang="ts">
  // Imports

  // Component Props interface
  interface Props {
    foo: string;
    bar: number;
    children?: Snippet | undefined; // <- children are sub-components that can be rendered inside a component
  }

  let { foo, bar, children = undefined }: Props = $props();

  // More functions, variables, logic, ...
</script>

<!--
@component
This is my new component.

Usage:

> \`\`\`svelte

Reviewer: Is it possible, that this will not render correctly? At least, my preview in web storm doesnt recognize it correctly.

Author: We can't render ``` inside a code block. I added the backslashes to symbolize that it's escaped.

    <MyNewComponent foo="Example" bar={0}>
        <span>This is a span!</span>
    </MyNewComponent>
\`\`\`
-->
<div>
  <h1>{foo}</h1>
  <span>{bar}</span>
  <div>{@render children?.()}</div>
</div>
```

> The [NavigationBar component](https://github.com/SE-UUlm/snowballr-frontend/blob/develop/src/lib/components/composites/navigation-bar/NavigationBar.svelte)

Reviewer: I would not agree with you, this component is missing the comment you described 😆

Author: I only thought about how it demonstrates using props and children. I added a comment to it (finally after 1 year).

is a good example of a well-structured component.

### How to Pass HTML Attributes to a custom component

Sometimes we want to set HTML attributes inside the component from outside when using it. For this we can change the

> Props interface. From:

Reviewer: ```suggestion
`Props` interface. From:
```
Or property interface
